web/packages/teleport/src/Navigation/NavigationItem.tsx

```diff
@@ -84,8 +84,18 @@ export function NavigationItem(props: NavigationItemProps) {
   const ctx = useTeleport();
   const { clusterId } = useStickyClusterId();
 
-  const { navigationItem, route, isLocked, lockedNavigationItem, lockedRoute } =
-    props.feature;
+  const {
+    navigationItem,
+    route,
+    isLocked,
+    lockedNavigationItem,
+    lockedRoute,
+    hideFromNavigation,
+  } = props.feature;
+
+  if (hideFromNavigation) {
+    return null;
+  }
```

Comment:
This will cause the `useCallback` below to conditionally render which we can't have. We need to move it below any hooks.  [its updated here](https://github.com/gravitational/teleport/pull/30596) if we end up merging those two together quickly, but should probably be fixed here before merge

Reply:
Wow, good catch. (Actually, I'd argue that this `useCallback` here is completely redundant, but let's skip it for now.)